docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+
+#### Alternatives
+
+- Split out health checks into a separate Policy: The upstream settings group is large and could be split into two Policies based on use cases. However, we want to limit the number of CRDs we maintain and reduce the overlap between CRDs. Two Policies that set similar directives and parameters in the upstream context could be complex to implement.
+
+### Authentication
+
+_Extension type:_ Inherited Policy
+
+_Resource type:_ CRD
+
+_Role(s):_ Cluster Operator, Application Developer
+
+_Extension point:_ Gateway, Gateway Listener, HTTPRoute Rule (once it's supported)
+
+_NGINX context(s):_ http, server, location
+
+OSS Features:
+
+- Basic Authentication (User/Password)
+- API Key Authentication
+
+OSS NGINX directives:
+
+- [`auth_basic`](https://nginx.org/en/docs/http/ngx_http_auth_basic_module.html#auth_basic)
+- [`auth_basic_user_file`](https://nginx.org/en/docs/http/ngx_http_auth_basic_module.html#auth_basic_user_file)
+
+NGINX Plus Features:
+
+- JSON Web Token (JWT) Authentication
+- OIDC Authentication
+
+NGINX Plus directives:
+
+- [`auth_jwt`](https://nginx.org/en/docs/http/ngx_http_auth_jwt_module.html#auth_jwt)
+- [`auth_jwt_type`](https://nginx.org/en/docs/http/ngx_http_auth_jwt_module.html#auth_jwt_type)
+- [`auth_jwt_key_file`](https://nginx.org/en/docs/http/ngx_http_auth_jwt_module.html#auth_jwt_key_file)
+- [`auth_jwt_key_request`](https://nginx.org/en/docs/http/ngx_http_auth_jwt_module.html#auth_jwt_key_request)
+
+#### Future Work
+
+- Add support for remote authentication or other authentication strategies.
+
+#### Alternatives
+
+- Begin with a Filter: Rather than beginning with an Inherited Policy, which is complex, we could start by adding an authentication Filter. This would allow us to roll out an authentication solution quickly to support the core use case for Application Developers. The use case for Cluster Operators applying authentication Policies is less well-known than the Application Developer's use case. Instead of leading with the Cluster Operator use case, we can begin with a Filter and wait for user feedback. Later, we can add a Policy and define how the Filter and Policy interact.
+- Wait for Gateway API GEP on authentication: There is an open [GEP issue](https://github.com/kubernetes-sigs/gateway-api/issues/1494) for authentication in the Gateway API repository. Instead of rolling out our own authentication solution, we could wait or even champion this GEP. The advantages of waiting are that if the GEP is implemented, we won't have to deprecate our bespoke authentication Policy in favor of the GEP's implementation, and we won't have to maintain an additional Policy. However, we don't know how long it will take the GEP to move from open to implementable. Authentication is a critical feature, and it may not be prudent to wait.
+
+### Observability
+
+_Extension type:_ Direct Policy
+
+_Resource type:_ CRD
+
+_Role(s):_ Application Developer
+
+_Extension point:_ HTTPRoute, HTTPRoute rule
```

Comment:
Is there a reason we'd want to explicitly not go after attaching the observability policy at the Gateway level?

We'll NEED the HTTPRoute and rule attachment, definitely. Was curious as to your thoughts around Gateway.

Reply:
In this case, we need to figure out if there's a use case for the ClusterOperator to provide sane defaults or enforce the following OTEL settings:

[otel_trace_context](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace_context)
[otel_span_name](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_name)
[otel_span_attr](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr)


The [otel_trace](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace) setting is different since it will also be exposed in the Gateway Settings. So the ClusterOperator will be able to enable/disable at the Gateway level.

Do you see a strong use case for the ClusterOperator to mess around with these settings? Seems like they are the responsibility of the App Dev. 

Due to their complexity, we should avoid Inherited Policies unless there is a really strong use case for defaults/overrides.